File: ChatAppClient/src/app/graphql/mutations/message.mutation.ts
import { gql } from 'apollo-angular';

export const SEND_MESSAGE = gql`
	mutation ($content: String!, $email: String!, $uniqueCodeGroup: String!) {
		sendMessage(
			model: { content: $content, email: $email, uniqueCodeGroup: $uniqueCodeGroup }
		) {
      content
      avatarSender
      senderName
      timeStamp
      senderId
		}
	}
`;

File: ChatAppServer/Schema/Mutations/MessageMutation.cs
using ChatAppServer.Models;
using ChatAppServer.Repositories.IRepositories;
using ChatAppServer.Schema.Types;
using ChatAppServer.Schema.Types.DTOTypes;

namespace ChatAppServer.Schema.Mutations
{
    [ExtendObjectType(typeof(Mutation))]
    public class MessageMutation
    {
        public async Task<MessageType> SendMessage(MessageTypeInput model, [Service] IMessageRepository messageRepository, [Service] IAuthRepository authRepository)
        {
            var user = await authRepository.GetUserByEmail(model.Email);

            var message = new Message
            {
                SenderId = user.Email!,
                SenderName = user.FullName,
                Content = model.Content,
                AvatarSender = user.AvatarUrl,
                TimeStamp = DateTime.Now
            };

            await messageRepository.SendMessage(message, model.UniqueCodeGroup);

            return new MessageType
            {
                SenderId = user.Email!,
                SenderName = user.FullName,
                Content = model.Content,
                AvatarSender = user.AvatarUrl,
            };
        }
    }
}
